src/openzaak/components/zaken/tests/test_zaak_filter_on_machtiging.py

```diff
@@ -0,0 +1,460 @@
+                            "id": "urn:etoegang:DV:00000001002308836000:services:9113",
+                            "uuid": "34085d78-21aa-4481-a219-b28d7f3282fc",
+                        }
+                    ],
+                },
+            },
+        )
+
+        with self.subTest(eHerkenningLevelOfAssurance.low_plus):
+            response = self.client.get(
+                self.url,
+                {"rol__machtiging__loa": eHerkenningLevelOfAssurance.low_plus},
+                **ZAAK_WRITE_KWARGS,
+            )
+
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+            self.assertEqual(response.json()["count"], 1)
+            self.assertEqual(
+                response.json()["results"][0]["url"],
+                f"http://testserver{reverse(zaak_low)}",
+            )
+
+        with self.subTest(eHerkenningLevelOfAssurance.high):
+            response = self.client.get(
+                self.url,
+                {"rol__machtiging__loa": eHerkenningLevelOfAssurance.high},
+                **ZAAK_WRITE_KWARGS,
+            )
+
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+            self.assertEqual(response.json()["count"], 2)
+
+        with self.subTest("invalid"):
+            response = self.client.get(
+                self.url,
+                {"rol__machtiging__loa": "invalid"},
+                **ZAAK_WRITE_KWARGS,
+            )
+
+            self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+            error = get_validation_errors(response, "rol__machtiging__loa")
+            self.assertEqual(error["code"], "invalid_choice")
+
+    def test_filter_rol_nnp_kvk(self):
+        zaaktype = ZaakTypeFactory.create()
+        roltype = RolTypeFactory.create(
+            zaaktype=zaaktype, omschrijving_generiek=RolOmschrijving.initiator
+        )
+        rol1, rol2, rol3 = RolFactory.create_batch(
+            3,
+            zaak__zaaktype=zaaktype,
+            roltype=roltype,
+            betrokkene_type=RolTypes.niet_natuurlijk_persoon,
+            roltoelichting="Created zaak",
+            contactpersoon_rol_naam="acting subject name",
+        )
+        NietNatuurlijkPersoon.objects.create(rol=rol1, kvk_nummer="12345678")
+        NietNatuurlijkPersoon.objects.create(rol=rol2, kvk_nummer="11122233")
+        NietNatuurlijkPersoon.objects.create(rol=rol3, inn_nnp_id="517439943")
+
+        response = self.client.get(
+            self.url,
+            {
+                "rol__betrokkeneIdentificatie__nietNatuurlijkPersoon__kvkNummer": "12345678"
+            },
+            **ZAAK_WRITE_KWARGS,
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.json()["count"], 1)
+        self.assertEqual(
+            response.json()["results"][0]["url"],
+            f"http://testserver{reverse(rol1.zaak)}",
+        )
+
+    def test_filter_rol_vestiging_kvk(self):
```

Comment:
These two tests don't seem specifically related to machtigen, so I'm not sure if they belong in this file

Reply:
These two tests check query params `rol__betrokkeneIdentificatie__nietNatuurlijkPersoon__kvkNummer` and `rol__betrokkeneIdentificatie__vestiging__kvkNummer`, which were added in this PR as experimental ones. 

I could move them to another file, but I've added them for machtiging purposes, so I've put them here

Reply:
hmm yeah I'm not sure where else to put them, let's leave them here